### PR TITLE
Fix OOM abusing large arrays (2x)

### PR DESCRIPTION
**Pull request by shargon**
Related to https://github.com/neo-project/neo-devpack-dotnet/issues/112
Apply https://github.com/neo-project/neo-devpack-dotnet/pull/110 on NEO2

### CODE REVIEW

**Comment by lightszero** [comment]
IsConstructor is for both cctor and ctor,
CctorSubVM is only for cctor

so contains “.cctor" is necessary

File: Neo.Compiler.MSIL/MSIL/Conv_Multi.cs [hunk 1]
```diff
@@ -1028,17 +1028,12 @@ private bool TryInsertMethod(NeoModule outModule, Mono.Cecil.MethodDefinition me
             try
             {
                 NeoMethod nm = new NeoMethod();
-                if (method.FullName.Contains(".cctor"))
+                if (method.IsConstructor)
```
Comment (lightszero) [inline Neo.Compiler.MSIL/MSIL/Conv_Multi.cs]: IsConstructor is for both cctor and ctor,
CctorSubVM is only for cctor

so contains “.cctor" is necessary

**Comment by shargon** [comment]
Then i will create a regex for this check

**Comment by shargon** [comment]
Travis is broken and I can't launch it again, master-2x doesn't appear to me in travis, @erikzhang could you restart travis?

**Comment by shargon** [comment]
Still broken

**Comment by shargon** [comment]
Could be merged? is the same protection as https://github.com/neo-project/neo-devpack-dotnet/pull/110 but for neo 2

**Comment by erikzhang** [comment]
Merged.

**Pull request merged**